src/PrestaShopBundle/Routing/LegacyRouterChecker.php

```diff
@@ -111,13 +114,29 @@ public function check(Request $request): bool
             // It's clearer to actually return a not found exception, for now the dispatcher is still used as fallback in index.php
             // but when it's cleared and only Symfony handles the whole routing then we can display a proper not found Symfony page
             if (!isset($controllers[strtolower($queryController)])) {
-                throw new NotFoundHttpException(sprintf('Unknown controller %s', $queryController));
+                // We need to define the languageId for context initialization in the legacy layout, because it will not be built
```

Comment:
```suggestion
                // We need to initialize the LanguageContext because it's used by components in the Symfony layout that will be displayed in the Not Found legacy page
```